SLO - Treat Absent Data for Window and Threshold Based SLOs - RN (#4217)
SumoLogic/sumologic-documentation: blog-service/2024-06-22-observability.md

---
title: Improved Data Handling for Window and Threshold-Based SLOs (Observability)
image: https://help.sumologic.com/img/sumo-square.png
keywords:
  - slo
  - observability
hide_table_of_contents: true
authors:
  - url: https://help.sumologic.com/release-notes-service/rss.xml
    image_url: /img/release-notes/rss-orange.png
---

We are happy to introduce a new update for handling missing data in window and threshold-based Service Level Objectives (SLOs). Previously, any window without data was treated as successful, and the threshold condition check was skipped. With this update, for windows without data, the threshold will be compared against **zero** to determine their success or failure. 

> This change simplifies expressing SLOs where missing data signifies unsuccessful windows. For example, consider an SLO definition where a non-zero number of heartbeat log messages is expected in a given window. Previously, such windows were treated as successful by default. With this update, modeling such cases will become much easier and more accurate.

Reviewer: ```suggestion
This change simplifies expressing SLOs where missing data signifies unsuccessful windows. For example, consider an SLO definition where absence of heartbeat log messages signifies an unsuccessful window. Previously, such windows were treated as successful by default. With this update, modeling such cases will become much easier and more accurate.
```

---

> title: SLO - Treat Absent Data for Window and Threshold Based SLOs (Observability)

Reviewer 2: ```suggestion
title: Improved Data Handling for Window and Threshold-Based SLOs (Observability)
```

Reviewer: +1

image: https://help.sumologic.com/img/sumo-square.png
keywords:
  - slo
  - observability
hide_table_of_contents: true
authors:
  - url: https://help.sumologic.com/release-notes-service/rss.xml
    image_url: /img/release-notes/rss-orange.png
---

> We are happy to introduce you to a new change in handling missing data for window and threshold-based Service Level Objectives (SLOs). Previously, any absent data in an interval was assumed to be in the goodCnt category. With this update, missing data will be treated as **zero**, ensuring a more precise determination of the data category after setting the parameter value. Below are some of the key advantages that this change offers:
> 
> **Enhanced Accuracy**. Treating missing data as 0 prevents incorrect assumptions about the data quality, leading to more reliable SLO evaluations.
> **Consistency**. Ensures uniformity in data treatment across different intervals, improving the overall consistency of SLO calculations.

Reviewer 2: ```suggestion
We are happy to introduce a new update for handling missing data in window and threshold-based Service Level Objectives (SLOs). Previously, any absent data in an interval was assumed to be in the `goodCnt` category. With this update, missing data will now be treated as **zero**, ensuring a more precise determination of the data category after setting the parameter value. Below are some of the key advantages of this change:

* **Enhanced Accuracy**. Treating missing data as zero prevents incorrect assumptions about data quality, leading to more reliable SLO evaluations.
* **Consistency**. Ensures uniformity in data treatment across different intervals, improving the overall consistency of SLO calculations.
```

Reviewer: `Previously, any window without data was treated as a successful window and the threshold condition check was skipped. With this update, the threshold will be compared against **zero** for such windows to determine whether they are successful or not.`

> This change will enable expressing SLOs where missing data represents unsuccessful windows with ease. For example, consider an SLO definition where one expects a non-zero number of heartbeat log messages to be present in a given window. Previously, such windows were treated as successful by default. But with this update, modelling such cases will become much easier.
> 

Reviewer 2: ```suggestion
This change simplifies expressing SLOs where missing data signifies unsuccessful windows. For example, consider an SLO definition where a non-zero number of heartbeat log messages is expected in a given window. Previously, such windows were treated as successful by default. With this update, modeling such cases will become much easier and more accurate.
```

Reviewer 2: @apoorv-garg , let me know what you think!

---
title: Improved Data Handling for Window and Threshold-Based SLOs (Observability)
image: https://help.sumologic.com/img/sumo-square.png
keywords:
  - slo
  - observability
hide_table_of_contents: true
authors:
  - url: https://help.sumologic.com/release-notes-service/rss.xml
    image_url: /img/release-notes/rss-orange.png
---

> We are happy to introduce a new update for handling missing data in window and threshold-based Service Level Objectives (SLOs). Previously, any window without data was treated as a successful window and the threshold condition check was skipped. With this update, for window without data the threshold will be compared against **zero** to determine whether they are successful or not. Below are some of the key advantages of this change:

Reviewer: ```suggestion
We are happy to introduce a new update for handling missing data in window and threshold-based Service Level Objectives (SLOs). Previously, any window without data was treated as a successful window and the threshold condition check was skipped. With this update, for windows without data the threshold will be compared against **zero** to determine whether they are successful or not. Below are some of the key advantages of this change:
```

> * **Enhanced Accuracy**. Treating missing data as zero prevents incorrect assumptions about data quality, leading to more reliable SLO evaluations.

Reviewer: Instead of these two points which are very generic, we can just give an example below.

`Consider an SLI definition where one expects a non-zero number of heartbeat log messages to be present in a given window. Previously such windows were treated as successful by default. Modelling such cases will become much easier with this change.`

Author: @apoorv-garg I guess the two advantages listed will explain "how modelling such cases will become easier". So, it okay to add example and those advantage points?

Reviewer: it's not about accuracy, it's just a change in behaviour that will simplify achieving certain additional use cases. And consistency across windows is also not a customer goal. IMO the example is enough to drive home the value proposition.

Reviewer: ```
This change will enable expressing SLIs where missing data represents unsuccessful windows with ease. As an example, consider an SLI definition where one expects a non-zero number of heartbeat log messages to be present in a given window. Previously such windows were treated as successful by default. Modelling such cases will become much easier with this change.
```
